### PR TITLE
Add matrix documentation

### PR DESCRIPTION
**Pull request by tinloaf**
**Description:**

This adds documentation for the new Matrix component, which essentially adds Matrix chat bot capabilities.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#13355

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/matrix.markdown [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  homeserver: https://matrix.org
+  username: "@my_matrix_user:matrix.org"
+  password: supersecurepassword
+  rooms:
+    - "#hasstest:matrix.org"
+  commands:
+    - word: my_command
+```
+
+Configuration variables:
```
Comment (frenck) [inline source/_components/matrix.markdown]: Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
@tinloaf, I've reviewed it and left you a comment.

Also, this PR has a merge conflict. Please resolve that one as well.

**Comment by tinloaf** [comment]
I've converted it to the new config tags and resolved the merge conflict. We should probably still figure out how to deal with lists-of-X as a config option types (e.g. list-of-map, which I use here for the `commands` option), but for now I guess this does the trick.

**Review by frenck — DISMISSED** [review]

**Comment by frenck** [comment]
Thank for your update @tinloaf. I've approved the PR.
Can be merged as soon as the parent PR gets merged.

**Review by frenck — DISMISSED** [review]

**Pull request merged**